frame/staking-rewards/src/test/mod.rs

```diff
@@ -594,6 +557,265 @@ fn test_split_postion() {
 	});
 }
 
+mod claim {
+	use super::*;
+
+	#[test]
+	fn should_reward_correct_amount_when_not_early() {
+		let staker = ALICE;
+		let amount = 100_500;
+		let duration_preset = ONE_HOUR;
+		let total_rewards = 100;
+		let total_shares = 200;
+		let claim = 50;
+
+		with_stake(
+			staker,
+			amount,
+			duration_preset,
+			total_rewards,
+			total_shares,
+			Some(claim),
+			|pool_id, stake_id, _, stake_duration, staked_asset_id| {
+				let second_in_milliseconds = 1000;
+				Timestamp::set_timestamp(
+					Timestamp::now()
+						.saturating_add(stake_duration.saturating_mul(second_in_milliseconds))
+						.saturating_add(second_in_milliseconds),
+				);
+
+				let rewards_pool = StakingRewards::pools(StakingRewards::pool_count())
+					.expect("rewards_pool expected");
+
+				assert_ok!(StakingRewards::claim(Origin::signed(staker), stake_id));
+				assert_eq!(balance(staked_asset_id, &staker), amount);
```

Comment:
It's a bit confusing to read this as it looks like the staked amount was returned to the staker, we may need a comment to clarify what this verifies,

Reply:
d479d9f6c964927b723c51c903a6edaa0f583f91